src/data-management/store.ts

```diff
@@ -0,0 +1,99 @@
+import { create } from 'zustand';
+import { StorageService } from './storage-service';
+import { Note } from './types';
+
+export interface NotesState {
+    notes: Note[];
+    trash: Note[];
+
+    addNote: (newNote: Pick<Note, 'title' | 'description'>) => void;
+    updateNote: (
+        id: string,
+        updateNote: Partial<Pick<Note, 'title' | 'description' | 'isPinned'>>
+    ) => void;
+    deleteNote: (id: string, permanently?: boolean) => void;
+    syncNotes: () => void;
+    syncTrash: () => void;
+    getNoteById: (id: string) => Promise<Note | undefined>;
+    restoreNote: (id: string) => Promise<void>;
+    pinNote: (id: string) => Promise<void>;
+    unPinNote: (id: string) => Promise<void>;
+}
+
+const storageService = new StorageService();
+
+export const useNoteStore = create<NotesState>((set) => {
```

Comment:
The last thing i guess.
Storage service should be injected as dependency into createNotesStore function